bee-network/bee-autopeering/src/server.rs

```diff
@@ -0,0 +1,290 @@
+                                peer_id,
+                            };
+
+                            // Depending on the message type, forward it to the appropriate manager.
+                            match msg_type as u32 {
+                                t if DISCOVERY_MSG_TYPE_RANGE.contains(&t) => {
+                                    discovery_tx.send(packet).expect("channel send error: discovery");
+                                }
+                                t if PEERING_MSG_TYPE_RANGE.contains(&t) => {
+                                    peering_tx.send(packet).expect("channel send error: peering");
+                                }
+                                _ => panic!("invalid message type"),
+                            }
+                        }
+                        Err(e) => {
+                            log::error!("UDP socket read error; stopping incoming packet handler. Cause: {}", e);
+                            // TODO: intiate shutdown of the system
+                            break 'recv;
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Runnable for OutgoingPacketHandler {
+    const NAME: &'static str = "OutgoingPacketHandler";
+    const SHUTDOWN_PRIORITY: u8 = 3;
+
+    type ShutdownSignal = ShutdownRx;
+
+    async fn run(self, mut shutdown_rx: Self::ShutdownSignal) {
+        let OutgoingPacketHandler {
+            outgoing_socket,
+            mut outgoing_rx,
+            local,
+            bind_addr,
+        } = self;
+
+        'recv: loop {
+            tokio::select! {
+                _ = &mut shutdown_rx => {
+                    break;
+                }
+                o = outgoing_rx.recv() => {
+                    if let Some(packet) = o {
+                        let OutgoingPacket {
+                            msg_type,
+                            msg_bytes,
+                            peer_addr,
+                        } = packet;
+
+                        if peer_addr == bind_addr {
+                            log::warn!("Trying to send to own bind address: {}. Ignoring packet.", peer_addr);
+                            continue 'recv;
+                        }
+
+                        let marshalled_bytes = marshal(msg_type, &msg_bytes);
+
+                        let signature = local.sign(&marshalled_bytes);
+                        let packet = Packet::new(msg_type, &marshalled_bytes, &local.public_key(), signature);
+
+                        let bytes = packet.to_protobuf().expect("error encoding outgoing packet");
+
+                        if bytes.len() > MAX_PACKET_SIZE {
+                            log::warn!("Trying to send too many bytes to {}. Ignoring...", peer_addr);
+                            continue 'recv;
+                        }
+
+                        let n = outgoing_socket.send_to(&bytes, peer_addr).await.expect("socket send error");
```

Comment:
I've read up on when `send_to` returns an error. It would occur iff a send happens to be `ipv4 -> ipv6` or `ipv6 -> ipv4`. Tbh I'ld rather have autopeering break forcing me to immediatedly fix it, rather than logging some error and keep the bug in the code for a while.

Reply:
Another reason is that the community running the node will much more likely report a bug if the node crashed. If it just logs an error, but the node continues to run, then they might think, that it's probably fine as long as it still runs.

Reply:
Closing this as I don't think the node should be kept alive if that `send` fails. The panic would indicate a real bug in the node which could go unnoticed otherwise. For `recv`s I absolutely agree with you (as we cannot be sure whether our peer is malfunctioning or flat out malicious), but for `send`s I think we should panic (because then it's our fault).